deploy/wmagent-venv-deploy.sh

```diff
@@ -0,0 +1,672 @@
+    echo "pythonCmd            : $pythonCmd and `which $pythonCmd`"
+    echo "======================================================="
+    echo -n "Continue? [y]: "
+    $assumeYes || read x && [[ $x =~ (n|no|nO|N|No|NO) ]] && return 102
+    echo "..."
+    echo "You still have 5 sec. to cancel before we proceed."
+    sleep 5
+}
+
+createVenv(){
+    # Function for creating minimal virtual environment. It uses global
+    # $venvCleanup to check if to clean the venv space before deployment.
+    # :param: None
+    echo
+    echo "======================================================="
+    echo "Creating minimal virtual environment:"
+    echo -n "Continue? [y]: "
+    $assumeYes || read x && [[ $x =~ (n|no|nO|N|No|NO) ]] && return 102
+    echo "..."
+
+    [[ -d $venvPath ]] || mkdir -p $venvPath || return $?
+    if $venvCleanup ; then
+        $pythonCmd -m venv --clear $venvPath || return $?
+    else
+        $pythonCmd -m venv $venvPath || return $?
+    fi
+}
+
+cloneWMCore(){
+    # Function for cloning WMCore source code and checkout to the proper branch
+    # or tag based on the script's runtime prameters.
+    # :param: None
+    echo
+    echo "======================================================="
+    echo "Cloning WMCore source code:"
+    echo -n "Continue? [y]: "
+    $assumeYes || read x && [[ $x =~ (n|no|nO|N|No|NO) ]] && return 101
+    echo "..."
+
+    wmSrcPath=${venvPath}/srv/WMCore           # WMCore source code target path
+
+    # NOTE: If the Virtual Environment is not to be cleaned during the current
+    #       deployment and we already have either a source directory synced from
+    #       previous deployments or a link at $wmSrcPath pointing to a source
+    #       directory outside the virtual env. we simply skip git actions to protect
+    #       developer's previous work.
+    if $noVenvCleanup && ( [[ -d $wmSrcPath ]] || [[ -h $wmSrcPath ]] ); then
+        echo "WMCore source has already been cloned and the NO Virtual Environment Cleanup is True."
+        return 101
+    else
+        [[ -d $wmSrcPath ]] ||  mkdir -p $wmSrcPath || return $?
+        cd $wmSrcPath
+        git clone $wmSrcRepo $wmSrcPath && git checkout $wmSrcBranch && [[ -n $wmTag ]] && git reset --hard $wmTag
+        cd -
+    fi
+}
+
+_pipUpgradeVenv(){
+    # Helper function used only for pip Upgrade for the current virtual env
+    # :param: None
+    cd $venvPath
+    # pip install $pipOpt wheel
+    # pip install $pipOpt --upgrade pip
+    pip install wheel
+    pip install --upgrade pip
+}
+
+activateVenv(){
+    # Function for activating the virtual environment
+    # :param: None
+    echo
+    echo "======================================================="
+    echo "Activate WMCore virtual env:"
+    echo -n "Continue? [y]: "
+    $assumeYes || read x && [[ $x =~ (n|no|nO|N|No|NO) ]] && return 102
+    echo "..."
+    source ${venvPath}/bin/activate
+    _pipUpgradeVenv
+}
+
+setupRucio(){
```

Comment:
Don't we get the `rucio-clients` package from the requirements of WMAgent?
On what concerns the rucio.cfg, isn't it shipped with the pypi wmagent package as well?

Reply:
It was actually even not called from the main function at all. Removed now.